File: launchdarkly/java/src/main/java/oseg/launchdarkly_examples/PostModelConfigExample.java
package oseg.launchdarkly_examples;

import com.launchdarkly.client.ApiException;
import com.launchdarkly.client.Configuration;
import com.launchdarkly.client.api.*;
import com.launchdarkly.client.auth.*;
import com.launchdarkly.client.JSON;
import com.launchdarkly.client.model.*;

import java.io.File;
import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.OffsetDateTime;
import java.util.ArrayList;
import java.util.List;
import java.util.Map;

public class PostModelConfigExample
{
    public static void main(String[] args)
    {
        var config = Configuration.getDefaultApiClient();
        ((ApiKeyAuth) config.getAuthentication("ApiKey")).setApiKey("YOUR_API_KEY");

        var modelConfigPost = new ModelConfigPost();
        modelConfigPost.id("id");
        modelConfigPost.key("key");
        modelConfigPost.name("name");
        modelConfigPost.icon("icon");
        modelConfigPost.provider("provider");
        modelConfigPost.tags(List.of (
            "tags",
            "tags"
        ));
        modelConfigPost.params(Map.of ());
        modelConfigPost.customParams(Map.of ());

        try
        {
            var response = new AiConfigsBetaApi(config).postModelConfig(
                "beta", // ldAPIVersion
                "default", // projectKey
                modelConfigPost
            );

            System.out.println(response);
        } catch (ApiException e) {
            System.err.println("Exception when calling AiConfigsBetaApi#postModelConfig");
            System.err.println("Status code: " + e.getCode());
            System.err.println("Reason: " + e.getResponseBody());
            System.err.println("Response headers: " + e.getResponseHeaders());
            e.printStackTrace();
        }
    }
}

File: launchdarkly/kotlin/src/main/kotlin/oseg/launchdarkly_examples/PostModelConfigExample.kt
package oseg.launchdarkly_examples

import com.launchdarkly.client.infrastructure.*
import com.launchdarkly.client.apis.*
import com.launchdarkly.client.models.*

import java.io.File
import java.time.LocalDate
import java.time.OffsetDateTime
import kotlin.collections.ArrayList
import kotlin.collections.List
import kotlin.collections.Map
import com.squareup.moshi.adapter

@ExperimentalStdlibApi
class PostModelConfigExample
{
    fun postModelConfig()
    {
        ApiClient.apiKey["Authorization"] = "YOUR_API_KEY"

        val modelConfigPost = ModelConfigPost(
            id = "id",
            key = "key",
            name = "name",
            icon = "icon",
            provider = "provider",
            tags = listOf (
                "tags",
                "tags",
            ),
            params = mapOf<String, Any> (),
            customParams = mapOf<String, Any> (),
        )

        try
        {
            val response = AIConfigsBetaApi().postModelConfig(
                ldAPIVersion = AIConfigsBetaApi.LdAPIVersionPostModelConfig.beta,
                projectKey = "default",
                modelConfigPost = modelConfigPost,
            )

            println(response)
        } catch (e: ClientException) {
            println("4xx response calling AIConfigsBetaApi#postModelConfig")
            e.printStackTrace()
        } catch (e: ServerException) {
            println("5xx response calling AIConfigsBetaApi#postModelConfig")
            e.printStackTrace()
        }
    }
}
